store/sqlstore/store.go

```diff
@@ -1422,19 +1422,26 @@ func (ss *SqlStore) migrate(direction migrationDirection) error {
 	return nil
 }
 
+var mysqlDSNRegex = regexp.MustCompile(
+	`^(?:(?P<user>.*?)(?::(?P<passwd>.*))?@)?` + // [user[:password]@]
+		`(?:(?P<net>[^\(]*)(?:\((?P<addr>[^\)]*)\))?)?` + // [net[(addr)]]
+		`\/(?P<dbname>.*?)` + // /dbname
+		`(?:\?(?P<params>[^\?]*))?$`) // [?param1=value1&paramN=valueN]
+
 func (ss *SqlStore) appendMultipleStatementsFlag(dataSource string) string {
 	// We need to tell the MySQL driver that we want to use multiStatements
 	// in order to make migrations work.
 	if ss.DriverName() == model.DATABASE_DRIVER_MYSQL {
-		u, err := url.Parse(dataSource)
-		if err != nil {
-			mlog.Critical("Invalid database url found", mlog.Err(err))
-			os.Exit(ExitGenericFailure)
+		match := mysqlDSNRegex.FindStringSubmatch(dataSource)
```

Comment:
Thoughts on using `mysql.ParseDSN` https://pkg.go.dev/github.com/go-sql-driver/mysql#ParseDSN here instead of hand-rolling the parser ourselves? And then do a [FormatDSN](https://pkg.go.dev/github.com/go-sql-driver/mysql#Config.FormatDSN) after that.